src/clockgen/clockgen.c

```diff
@@ -253,7 +253,7 @@ s32 _sceClockgenModuleStart(SceSize args __attribute__((unused)), void *argp __a
 }
 
 //0x00000320
-s32 _sceClockgenModuleRebootBefore(SceSize args __attribute__((unused)), void *argp __attribute__((unused)))
+s32 _sceClockgenModuleRebootBefore()
```

Comment:
The argument list of `module_reboot_before` is different depending on whether the module is a user module or a kernel module. A module is a kernel module when its module attribute contains the `SCE_MODULE_KERNEL` flag. This is the case for the kernel modules we are reverse engineering as part of uofw:
https://github.com/uofw/uofw/blob/276cb776451721f77775d71bee9d60bf17b07ba3/src/clockgen/clockgen.c#L31

For such kernel modules, `module_reboot_before` takes *four* arguments: https://github.com/uofw/uofw/blob/276cb776451721f77775d71bee9d60bf17b07ba3/src/modulemgr/rebootModule.c#L198

However, if the module is a user module, then its `module_reboot_before` function follows the standard thread entry function:
https://github.com/uofw/uofw/blob/276cb776451721f77775d71bee9d60bf17b07ba3/src/modulemgr/rebootModule.c#L88

Where `SceKernelThreadEntry` is defined as:
https://github.com/uofw/uofw/blob/276cb776451721f77775d71bee9d60bf17b07ba3/include/threadman_kernel.h#L13

With that said, we should update our `SCE_MODULE_REBOOT_BEFORE` macro to:
```c++
#define SCE_MODULE_REBOOT_BEFORE(name)  int module_reboot_before(void *arg0, s32 arg1, s32 arg2, s32 arg3) __attribute__((alias(name)))

#define SCE_MODULE_REBOOT_BEFORE_FOR_USER(name)  int module_reboot_before(SceSize arglen, void *argp) __attribute__((alias(name)))
```

Most of our reversed modules will then use the same macro as today. We have at least one module in uofw which is *not* a kernel module (UserSystemLib) but that one is not using `module_reboot_before` so nothing to change there.